tfx/components/example_gen/utils.py

```diff
@@ -300,46 +367,60 @@ def _retrieve_latest_span_version(
             'Cannot find %s number from %s based on %s' %
             (VERSION_PROPERTY_NAME, file_path, split_regex_pattern))
 
-    if latest_span is None or span_int > int(latest_span):
+    if latest_span_int is None or span_int > latest_span_int:
       # Uses str instead of int because of zero padding digits.
-      latest_span = span_str
+      latest_span_elems = span_strs
+      latest_span_int = span_int
       latest_version = version_str
-    elif (span_int == int(latest_span) and
+    elif (latest_span_int == span_int and
           (latest_version is None or version_int >= int(latest_version))):
       latest_version = version_str
 
-  if latest_span is None or (is_match_version and latest_version is None):
+  if latest_span_int is None or (is_match_version and
+                                 latest_version is None):
     raise ValueError('Cannot find matching for split %s based on %s' %
                      (split.name, split.pattern))
 
-  return latest_span, latest_version
+  # Update split pattern so executor can find the files to ingest.
+  if is_match_span:
+    split.pattern = split.pattern.replace(SPAN_SPEC, latest_span_elems[0])
+  elif is_match_date:
+    for spec, value in zip(DATE_SPECS, latest_span_elems):
+      split.pattern = split.pattern.replace(spec, value)
+
+  latest_version_int = None
+  if is_match_version:
+    split.pattern = split.pattern.replace(VERSION_SPEC, latest_version)
+    latest_version_int = int(latest_version)
+
+  return latest_span_int, latest_version_int
 
 
 def calculate_splits_fingerprint_span_and_version(
     input_base_uri: Text, splits: Iterable[example_gen_pb2.Input.Split]
-) -> Tuple[Text, Text, Optional[Text]]:
+) -> Tuple[Text, int, Optional[int]]:
   """Calculates the fingerprint of files in a URI matching split patterns.
 
-  If a pattern has the {SPAN} placeholder and, optionally, the {VERSION}
-  placeholder, attempts to find aligned values that results in all splits
-  having the most recent span and most recent version for that span.
+  If a pattern has the {SPAN} placeholder or the Date spec placeholders, {YYYY},
+  {MM}, and {DD}, and optionally, the {VERSION} placeholder, attempts to find
+  aligned values that results in all splits having the most recent span and most
+  recent version for that span.
 
   Args:
     input_base_uri: The base path from which files will be searched.
-    splits: An iterable collection of example_gen_pb2.Input.Split objects. Note
-      that this function will update the {SPAN} in this and {VERSION} tags in
-      the split config to actual Span and Version numbers.
+    splits: An iterable collection of example_gen_pb2.Input.Split objects.
```

Comment:
add the Note the tags will be updated part in the previous comment

Reply:
updated, thanks!